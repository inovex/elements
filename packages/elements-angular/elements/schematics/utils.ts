import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
