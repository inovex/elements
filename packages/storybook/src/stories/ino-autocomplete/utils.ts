import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
