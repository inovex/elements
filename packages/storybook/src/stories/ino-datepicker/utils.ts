import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
