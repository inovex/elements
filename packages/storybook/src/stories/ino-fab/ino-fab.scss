.story-fab {
  .ino-fab-variants {
    display: flex;
    justify-content: space-around;
  }
}
