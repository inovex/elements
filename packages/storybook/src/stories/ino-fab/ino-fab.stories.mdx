import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-fab/readme.md";

<Meta title="Buttons/ino-fab/Documentation" />

<Description markdown={readme} />
