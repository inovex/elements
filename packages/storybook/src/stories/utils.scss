/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
