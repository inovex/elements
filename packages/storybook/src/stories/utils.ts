import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
