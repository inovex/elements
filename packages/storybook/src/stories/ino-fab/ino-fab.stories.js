import { storiesOf } from '@storybook/html';
import { text, boolean, select } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-fab/readme.md';

import ICONS from '_local-elements/src/components/ino-icon/icons';
import './ino-fab.scss';

export default {
  title: 'Buttons/<ino-fab>',
  decorators: [withStencilReadme(componentReadme)],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-fab">
    <h4>Customizable Fab</h4>
     <style>
          ino-fab.customizable-fab {
            --ino-fab-color: ${text('--ino-fab-color', '#ffffff', 'Custom Properties')};
            --ino-fab-background-color: ${text('--ino-fab-background-color', '#3d40f5', 'Custom Properties')};
            --ino-fab-background-color-hover: ${text('--ino-fab-background-color-hover', '#5d60f7', 'Custom Properties')};
            --ino-fab-background-color-active: ${text('--ino-fab-background-color-active', '#0d10f3', 'Custom Properties')};
            --ino-fab-icon-color: ${text('--ino-fab-icon-color', '#ffffff', 'Custom Properties')};
            --ino-fab-color-disabled: ${text('--ino-fab-color-disabled', '#ffffff', 'Custom Properties')};
            --ino-fab-background-color-disabled: ${text('--ino-fab-background-color-disabled', '#9d9d9d', 'Custom Properties')};
            --ino-fab-icon-color-disabled: ${text('--ino-fab-icon-color-disabled', '#ffffff', 'Custom Properties')};
          }
      </style>
      <ino-fab
        class="customizable-fab"
      ino-edge-position="${select(
        'ino-edge-position',
        ['top-left', 'top-right', 'bottom-right', 'bottom-left', 'none'],
        'top-left'
      )}"
      ino-icon="${select('ino-leading-icon', ICONS, 'add')}"
      ino-label="${text('ino-label', 'Label')}"
      ino-extended="${boolean('ino-extended', false)}"
      ino-disabled="${boolean('ino-disabled', false)}"
      ino-mini="${boolean('ino-mini', false)}"></ino-fab>

    <h4>Variants</h4>
      <div class="ino-fab-variants">
        <ino-fab ino-tooltip-placement="none" ino-label="Default Fab" ino-icon="add"></ino-fab>
        <ino-fab ino-mini ino-label="Mini Fab" ino-icon="add"></ino-fab>
        <ino-fab ino-extended ino-label="Extended Button" ino-icon="add"></ino-fab>
        <ino-fab ino-disabled ino-label="Disabled Button" ino-icon="add"></ino-fab>
      </div>
    </div>
    <h4>Different Edge Positions</h4>
    <div class="ino-fab-variants">
       <ino-fab ino-label="top-left" ino-icon="add" ino-edge-position="top-left"></ino-fab>
       <ino-fab ino-label="top-right" ino-icon="add" ino-edge-position="top-right"></ino-fab>
       <ino-fab ino-label="bottom-right" ino-icon="add" ino-edge-position="bottom-right"></ino-fab>
       <ino-fab ino-label="bottom-left" ino-icon="add" ino-edge-position="bottom-left"></ino-fab>
       <ino-fab ino-label="none" ino-icon="add" ino-edge-position="none"></ino-fab>
    </div>
  `;
