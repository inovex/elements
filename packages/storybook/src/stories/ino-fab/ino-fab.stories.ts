import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator, withIconControl } from '../utils';

import './ino-fab.scss';

export default {
  title: 'Buttons/<ino-fab>',
  component: 'ino-fab',
  decorators: [defaultDecorator],
};

export const Playground: Story<Components.InoFab> = (args) => html`
  <ino-fab
    class="customizable-fab"
    disabled="${args.disabled}"
    edge-position="${args.edgePosition}"
    extended="${args.extended}"
    label="${args.label}"
    mini="${args.mini}"
    tooltip-placement="${args.tooltipPlacement}"
  >
    <ino-icon slot="icon-leading" icon="${args.icon}" />
  </ino-fab>
`;

Playground.args = {
  disabled: false,
  edgePosition: 'top-left',
  extended: false,
  label: 'Label',
  mini: false,
  tooltipPlacement: 'right',
};

withIconControl(Playground, 'icon', 'add');

Playground.argTypes.edgePosition = {
  control: {
    type: 'select',
  },
  options: ['top-left', 'top-right', 'bottom-right', 'bottom-left', 'none'],
};

Playground.argTypes.tooltipPlacement = {
  control: {
    type: 'select',
  },
  options: [
    'auto',
    'auto-end',
    'auto-start',
    'bottom',
    'bottom-end',
    'bottom-start',
    'left',
    'left-end',
    'left-start',
    'none',
    'right',
    'right-end',
    'right-start',
    'top',
    'top-end',
    'top-start',
  ],
};

export const Variants = () => html`
  <div class='ino-fab-variants'>
    <ino-fab tooltip-placement='none' label='Default Fab'>
      <ino-icon slot='icon-leading' icon='add'></ino-icon>
    </ino-fab>
    <ino-fab mini label='Mini Fab'>
      <ino-icon slot='icon-leading' icon='add'></ino-icon>
    </ino-fab>
    <ino-fab extended label='Extended Button'>
      <ino-icon slot='icon-leading' icon='add'></ino-icon>
    </ino-fab>
    <ino-fab disabled label='Disabled Button'>
      <ino-icon slot='icon-leading' icon='add'></ino-icon>
    </ino-fab>
  </div>
  </div>
`;

export const EdgePositions = () => html`
  <div class="ino-fab-variants">
    <ino-fab label="top-left" edge-position="top-left">
      <ino-icon slot="icon-leading" icon="add"></ino-icon>
    </ino-fab>
    <ino-fab label="top-right" edge-position="top-right">
      <ino-icon slot="icon-leading" icon="add"></ino-icon>
    </ino-fab>
    <ino-fab label="bottom-right" edge-position="bottom-right">
      <ino-icon slot="icon-leading" icon="add"></ino-icon>
    </ino-fab>
    <ino-fab label="bottom-left" edge-position="bottom-left">
      <ino-icon slot="icon-leading" icon="add"></ino-icon>
    </ino-fab>
    <ino-fab label="none" edge-position="none">
      <ino-icon slot="icon-leading" icon="add"></ino-icon>
    </ino-fab>
  </div>
`;
