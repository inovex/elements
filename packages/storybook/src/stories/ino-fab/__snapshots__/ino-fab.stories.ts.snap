// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/<ino-fab> EdgePositions smoke-test 1`] = `
<div id="root-inner">
  <div class="story-fab">
    <div class="ino-fab-variants">
      <ino-fab label="top-left"
               edge-position="top-left"
               class="ino-fab--edge-position-top-left hydrated"
               id="fab-helper-text__5"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab label="top-right"
               edge-position="top-right"
               class="ino-fab--edge-position-top-right hydrated"
               id="fab-helper-text__6"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab label="bottom-right"
               edge-position="bottom-right"
               class="ino-fab--edge-position-bottom-right hydrated"
               id="fab-helper-text__7"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab label="bottom-left"
               edge-position="bottom-left"
               class="ino-fab--edge-position-bottom-left hydrated"
               id="fab-helper-text__8"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab label="none"
               edge-position="none"
               class="ino-fab--edge-position-none hydrated"
               id="fab-helper-text__9"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-fab> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-fab">
    <ino-fab class="customizable-fab ino-fab--edge-position-top-left hydrated"
             disabled="false"
             edge-position="top-left"
             extended="false"
             label="Label"
             mini="false"
             tooltip-placement="right"
             id="fab-helper-text__0"
    >
      <ino-icon slot="icon-leading"
                icon="add"
                class="hydrated ino-icon--color-scheme-primary"
      >
      </ino-icon>
    </ino-fab>
  </div>
</div>
`;

exports[`Buttons/<ino-fab> Variants smoke-test 1`] = `
<div id="root-inner">
  <div class="story-fab">
    <div class="ino-fab-variants">
      <ino-fab tooltip-placement="none"
               label="Default Fab"
               class="ino-fab--edge-position-top-left hydrated"
               id="fab-helper-text__1"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab mini
               label="Mini Fab"
               class="ino-fab--edge-position-top-left hydrated"
               id="fab-helper-text__2"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab extended
               label="Extended Button"
               class="ino-fab--edge-position-top-left hydrated"
               id="fab-helper-text__3"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
      <ino-fab disabled
               label="Disabled Button"
               class="ino-fab--edge-position-top-left hydrated"
               id="fab-helper-text__4"
      >
        <ino-icon slot="icon-leading"
                  icon="add"
                  class="ino-icon--color-scheme-primary hydrated"
        >
        </ino-icon>
      </ino-fab>
    </div>
  </div>
</div>
`;
